File: src/python/finished/missing_number.py
from typing import List

# https://leetcode.com/problems/missing-number/


class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        ans = 0
        for i in range(len(nums) + 1):
            ans ^= i ^ (nums[i] if i < len(nums) else 0)
        return ans

File: src/rust/finished/missing_number.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut ans = 0;
        for i in 0..=nums.len() {
            ans ^= i as i32;
            if i < nums.len() {
                ans ^= nums[i];
            }
        }
        ans
    }
}
